portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/PageHandler.java

```diff
@@ -257,6 +257,17 @@ public void includePage(PortalRenderContext rcontext, HttpServletResponse res,
 				for (Iterator i = tools.iterator(); i.hasNext();)
 				{
 					ToolConfiguration placement = (ToolConfiguration) i.next();
+
+					if (site != null)
```

Comment:
I realise that this is being copied from line 229 but `SitePage.getContainingPage()` would have seemed to have been the way to get the Site and I don't think this should ever be null.